lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -531,7 +541,9 @@ def get_path(self):
     def do_3d_projection(self):
         s = self._segment3d
         if self._axlim_clip:
-            xs, ys, zs = _viewlim_mask(*zip(*s), self.axes)
+            mask = _viewlim_mask(*zip(*s), self.axes)
+            xs, ys, zs = np.ma.array(zip(*s),
```

Comment:
I'm not clear how large s is. Does it make sense to store `zip(*s)` in a variable and reuse it instead of calculating it twice?

Reply:
s for a single Patch or PathPatch should be small. But the main reason for using zip repeatedly here is that s might be a masked or unmasked array, so we're avoiding some type checking and casting by just using the variable directly.